generic/ras_extended.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Pavithra <pavrampu@linux.vnet.ibm.com>
+
+import os
+from shutil import copyfile
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado import skipIf
+from avocado.utils.software_manager import SoftwareManager
+
+IS_POWER_NV = 'PowerNV' in open('/proc/cpuinfo', 'r').read()
+IS_KVM_GUEST = 'qemu' in open('/proc/cpuinfo', 'r').read()
+
+
+class RASTools(Test):
+
+    """
+    This test checks various RAS tools:
+    """
+    is_fail = 0
+
+    def run_cmd(self, cmd):
+        cmd_result = process.run(cmd, ignore_status=True, sudo=True,
+                                 shell=True)
+        if cmd_result.exit_status != 0:
+            self.is_fail += 1
+        return
+
+    def setUp(self):
+        architecture = os.uname()[4]
+        if "ppc" not in architecture:
+            self.skip("supported only on Power platform")
+        sm = SoftwareManager()
+        for package in ("ppc64-diag", "powerpc-utils", "lsvpd", "sysfsutils"):
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error("Fail to install %s required for this"
+                           " test." % package)
+
+    @skipIf(IS_POWER_NV or IS_KVM_GUEST, "This test is not supported on KVM guest or PowerNV platform")
+    def test1_uesensor(self):
+        self.log.info("===============Executing uesensor tool test===="
+                      "===========")
+        self.run_cmd("uesensor -l")
+        self.run_cmd("uesensor -a")
+        if self.is_fail >= 1:
+            self.fail("%s command(s) failed in uesensor tool "
+                      "verification" % self.is_fail)
+
+    @skipIf(IS_POWER_NV or IS_KVM_GUEST, "This test is not supported on KVM guest or PowerNV platform")
+    def test1_serv_config(self):
+        self.log.info("===============Executing serv_config tool test===="
+                      "===========")
+        self.run_cmd("serv_config -l")
+        self.run_cmd("serv_config -b")
+        self.run_cmd("serv_config -s")
+        self.run_cmd("serv_config -r")
+        self.run_cmd("serv_config -m")
+        self.run_cmd("serv_config -d")
+        self.run_cmd("serv_config --remote-maint")
+        self.run_cmd("serv_config --surveillance")
+        self.run_cmd("serv_config --reboot-policy")
+        self.run_cmd("serv_config --remote-pon")
+        self.run_cmd("serv_config -d --force")
```

Comment:
If possible can you please use list for different command options and iterate over. Thanks!!